table_engine/src/partition/rule/df_adapter/mod.rs

```diff
@@ -0,0 +1,63 @@
+// Copyright 2022 CeresDB Project Authors. Licensed under Apache-2.0.
+
+//! Partition rule datafusion adapter
+
+use common_types::{row::RowGroup, schema::Schema};
+use datafusion_expr::Expr;
+
+use self::extractor::KeyExtractor;
+use super::factory::PartitionRuleFactory;
+use crate::partition::{
+    rule::{df_adapter::extractor::FilterExtractorRef, factory::PartitionRuleRef},
+    BuildPartitionRule, PartitionInfo, Result,
+};
+
+pub(crate) mod extractor;
+
+/// Partition rule's adapter for datafusion
+pub struct DfPartitionRuleAdapter {
+    /// Partition rule
+    rule: PartitionRuleRef,
+
+    /// `PartitionFilter` extractor for datafusion `Expr`
+    extractor: FilterExtractorRef,
+}
+
+impl DfPartitionRuleAdapter {
```

Comment:
Add some tests for DfPartitionRuleAdapter.

Reply:
Actually, I think tests of interfaces are not suitable to add here.
It should be placed in the detailed implementation module.
But just left them now?